### PR TITLE
✨ feat: adding Arabic Language Support

### PR DESCRIPTION
**Pull request by MEClouds**
#### 💻 变更类型 | Change Type

- [x] ✨ feat (New features related to Arabic language support)
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [x] 💄 style (Styling changes for better readability or aesthetics, including RTL support)
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

This pull request introduces Arabic language support to the Lobe Chat project, following the guidelines provided in the [Add New Locale documentation](https://github.com/lobehub/lobe-chat/wiki/Add-New-Locale).

#### 📝 补充信息 | Additional Information
I have added right-to-left (RTL) style changes to enhance the visual presentation for the Arabic language.

### Screenshots
image

image

### CODE REVIEW

File: src/locales/resources.ts [hunk 1]
```diff
@@ -41,6 +42,10 @@ type LocaleOptions = {
 }[];
 
 export const localeOptions: LocaleOptions = [
+  {
+    label: 'Arabic',
```
Comment (arvinxx) [inline src/locales/resources.ts]: Please change to native Arabic lang here

**Review by arvinxx — CHANGES_REQUESTED** [review]
@MEClouds Thanks for your amazing work! I have check your code, it's great!  Just need a few improvements:

1. There is a FOUC when using Arabic with server side rendering:
image

the fix method is to add the `dir` attribute to the `html` tag:
https://github.com/lobehub/lobe-chat/blob/main/src/app/layout.tsx#L25

```tsx

const direction = isRtlLang(lng?.value) ? 'rtl' : 'ltr';

return <html lang={lang?.value || DEFAULT_LANG} dir={direction} >
```
2. Please use native Arabic display here for consistency.
image

**Comment by MEClouds** [comment]
Thank you for your kind words! I'm thrilled to hear that you've reviewed the code and found it great. I appreciate your feedback and I'm ready to implement the suggested improvements.

**Review by arvinxx — APPROVED** [review]
Great Job! LobeChat now are more user friendly with Arabic Locale. I think we can merge this PR.

And I think there are some more improvement we should work with `lobe-ui`

like `SearchBar` and `DraggablePanel`

image


image

**Pull request merged**